code/modules/recycling/disposal.dm

```diff
@@ -724,8 +722,7 @@
 		if(H && H.loc)
 			expel(H, T, 0)
 
-	spawn(2) //Delete pipe after 2 ticks to ensure expel proc finished
-		qdel(src)
+	QDEL_IN(src, 2) //Delete pipe after 2 ticks to ensure expel proc finished
```

Comment:
```suggestion
	QDEL_IN(src, 0.2 SECONDS) //Delete pipe after 2 ticks to ensure expel proc finished
```

Reply:
Read the comment.

Reply:
Isn't this behavior being changed, though, from a hard 2 ticks to a more variable standard by the callback?

Reply:
it does what the comment says.